SSA/Projects/InstCombine/LLVM/Enumerator.lean

```diff
@@ -115,40 +127,84 @@ def icmpRows (pred : LLVM.IntPredicate) : Array Row := Id.run do
         rows := rows.push row
   rows
 
-def main : IO Unit := do
+
+def testNameToRowName : String → String
+  | testName => match testName.splitOn "_" with
+    | [_,name] => name
+    | [_,name, op] => s!"{name}.{op}"
+    | _ => "Error"
+
+/-- Produce rows for a `ConcreteCliTest` -/
+def concreteCliTestRows (test : ConcreteCliTest) : IO <| Array Row := do
+  let mut rows := #[]
+  let argTys : List test.code.getTy := test.context.reverse -- reverse because context is stack
+  let mut args : Array (Array (Option ℤ)) := #[]
+  for ty in argTys do
+    match ty with
+      | .bitvec (.concrete w) => args := args.push <| (BitVec.inputsForWidth w).map (Option.map BitVec.toInt)
```

Comment:
Could we write this as `(List.map ...).toArray`?
That seems a bit cleaner than going full-imperative, although I wonder if there is a performance difference